File: app/src/main/java/com/esorokin/lantector/ui/plugins/AlertMessagePlugin.java
package com.esorokin.lantector.ui.plugins;

import android.support.v7.app.AlertDialog;
import android.text.TextUtils;

import com.esorokin.lantector.R;
import com.esorokin.lantector.ui.plugins.base.BaseDependencyPlugin;

public class AlertMessagePlugin extends BaseDependencyPlugin<ContextProvider> {
	private AlertDialog dialog;

	public AlertMessagePlugin(ContextProvider dependency) {
		super(dependency);
	}

	@Override
	public void onDestroy() {
		super.onDestroy();
		hideMessage();
	}

	public void showMessage(String title, MessageHideListener hideListener) {
		showMessage(title, "", getDependency().getContext().getString(R.string.ok), hideListener);
	}

	public void showMessage(String title, String message, String okText, MessageHideListener hideListener) {
		hideMessage();

		AlertDialog.Builder builder = new AlertDialog.Builder(getDependency().getContext());
		if (!TextUtils.isEmpty(title)) {
			builder.setTitle(title);
		}

		if (!TextUtils.isEmpty(message)) {
			builder.setMessage(message);
		}

		if (!TextUtils.isEmpty(okText)) {
			builder.setPositiveButton(okText, (dialogInterface, i) -> {
				if (hideListener != null) {
					hideListener.hideMessage();
				}
			});
		}

		dialog = builder.show();
	}

	public void hideMessage() {
		if (dialog != null) {
			dialog.dismiss();
		}
	}

	public interface MessageHideListener {
		void hideMessage();
	}
}

File: app/src/main/java/com/esorokin/lantector/ui/plugin/AlertMessagePlugin.kt
package com.esorokin.lantector.ui.plugin

import android.content.Context
import android.support.v7.app.AlertDialog
import com.esorokin.lantector.R
import com.esorokin.lantector.ui.plugin.base.BasePlugin

class AlertMessagePlugin(private val contextProvider: () -> Context) : BasePlugin() {
    private var dialog: AlertDialog? = null

    override fun onDestroy() {
        super.onDestroy()
        hideMessage()
    }

    fun showMessage(message: String? = null,
                    title: String? = null,
                    okText: String = contextProvider.invoke().getString(R.string.ok),
                    hideListener: () -> Unit = {}) {
        hideMessage()

        val builder = AlertDialog.Builder(contextProvider.invoke())

        title?.let { builder.setTitle(it) }
        message?.let { builder.setMessage(it) }
        builder.setPositiveButton(okText, { _, _ -> hideListener.invoke() })
        builder.setCancelable(true)
        builder.setOnCancelListener { hideListener.invoke() }

        dialog = builder.show()
    }

    fun hideMessage() {
        dialog?.dismiss()
    }
}